src/commands/build.rs

```diff
@@ -29,10 +35,12 @@ pub struct BuildCommand {
 
 impl BuildCommand {
     pub async fn run(self) -> Result<()> {
-        let manifest_file = self
-            .app
-            .as_deref()
-            .unwrap_or_else(|| DEFAULT_MANIFEST_FILE.as_ref());
+        let manifest_file = match (self.app_source.as_deref(), self.file_source.as_deref()) {
+            (None, None) => DEFAULT_MANIFEST_FILE.as_ref(),
+            (Some(p), None) => p,
+            (None, Some(p)) => p,
+            (Some(_), Some(_)) => bail!("Cannot specify more than one application"),
+        };
```

Comment:
Is there a reason we don't need the app source type in the build command? Does `spin_build::build(manifest_file)` just out of the box support bindle/registry sources?

Reply:
@calebschoepp Remote sources are already built (and usually wouldn't carry around their source code).  It's not meaningful to pass a remote reference to `spin build`, only a manifest file.  Which is why it currently accepts only the `-f` flag, and in a flagless world could only ever accept the `File` source type, making the option pointless.

I'm not sure I've understood the question correctly - if this doesn't answer it, hound me.

Reply:
Nope, totally on the mark. Thanks for answering my question. I didn't put two and two together that a registry source would already be built.